src/cargo/core/shell.rs

```diff
@@ -415,7 +415,11 @@ impl Shell {
         writeln!(
             self.err(),
             "{}",
-            Renderer::styled().term_width(term_width).render(message)
+            Renderer::styled()
+                .note(NOTE)
```

Comment:
this should be a pr to annotate snippets if there is a discrepancy

Reply:
Oh, I see. I thought the CI test failure was a false positive:

  
  ```
       Summary semver requires new major version: 1 major and 0 minor checks failed
  --- failure constructible_struct_adds_field: externally-constructible struct adds field ---
  
  Description:
  A pub struct constructible with a struct literal has a new pub field. Existing struct literals must be updated to include the new field.
          ref: https://doc.rust-lang.org/reference/expressions/struct-expr.html
         impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.41.0/src/lints/constructible_struct_adds_field.ron
```

but there's also a UI test failing.

Reply:
The expectation is that `annotate-snippets` defaults *are* rustc's defaults.  If `annotate-snippets` is not matching rustc, then that is a bug with `annotate-snippers` that should be resolved.  We should not work around that bug in Cargo.